### PR TITLE
Filter Command (by tag)

### PR DESCRIPTION
**Pull request by chenyixin0**
Current commits (up to commit 2 - bug fixes) only implement filter by tags
filter by date can wait until next week I guess

### CODE REVIEW

**Comment by chenyixin0** [comment]
force-push because previous push was not meant for this PR and included a lot of buggy code

**Comment by eunrcn** [comment]
dear evan, 
thank you for adding the filter by tags command. you said its is simple but im sure its not, you are just too good of a coder, to be able to score so well for 2040 takes skill okay. on my first day of class (i skipped the first class on tuesday so im refering to friday) i sat beside ashley and across from you. from the moment the teacher announced that we will have another group, i knew i had to group with you, cause you're obviously the coolest person in the room. and you don't know how happy i was when i very calmly asked ashley to group with me and she replied "wait ah let me ask my friend yixin" because ofc i was excited to work with you, and knowing that the two of yall knew it each other was perfect. i do hope to spend more time with you and the team, time always passes so fast when im with y'all.

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        return predicate.equals(otherFilterCommand.predicate);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("predicate", predicate)
+                .toString();
+    }
+}
```
Comment (Jawad280) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:61]: Overall, the implementation of the Filter Command is neat and clean 👍🏾

File: src/main/java/seedu/address/model/person/MatchingTagPredicate.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+        MatchingTagPredicate otherMatchingTagPredicate = (MatchingTagPredicate) other;
+        return tag.equals(otherMatchingTagPredicate.tag);
+    }
+
+    @Override
+    public String toString() {
+        return "tag " + tag;
+    }
+}
```
Comment (Jawad280) [inline src/main/java/seedu/address/model/person/MatchingTagPredicate.java:40]: I like how you separated the matching tags as a class to provide a clearer and efficient implementation.
Looks Good 👍🏾

**Review by Jawad280 — COMMENTED** [review]
Generally great job in implementing the Filter Command ! The code was clean and easy to follow. Good job on working out the testcases as well ! LGTM 👍🏾

**Pull request merged**